### DISCUSSION TITLE
Want to retrieve user to asset details from CS IDP

### DISCUSSION THREAD

**Discussion by ninja5244** [CrowdStrike/psfalcon#373] [Q&A] [answered]
Tried to retrieve user to asset details using Get-FalconHost -Detailed -Filter  -Include login_history | select hostname,last_login_user,login_history. but i'm getting machine login history and some of the machines doesn't have login details. Is there any PSfalcon command to retrieve user to asset details?

**Comment by bk-cs** [comment, marked as answer]
`login_history` (provided by `Get-FalconHost -Include login_history`  or `Get-FalconHost -Login`) comes from an API that is device-specific and not related to Falcon Identity Protection. That data comes from the Identity Graph, which is searched using GraphQL queries and [Invoke-FalconIdentityGraph](https://github.com/crowdstrike/psfalcon/wiki/Invoke-FalconIdentityGraph).
